app/controllers/admin/page_elements_controller.rb

```diff
@@ -0,0 +1,59 @@
+# encoding:UTF-8
+class Admin::PageElementsController < ApplicationController
+  before_action :load_permissions
+  load_and_authorize_resource :page, parent: true, find_by: :url
+  load_and_authorize_resource :page_element, through: :page
+
+  def index
+    page = Page.includes(:page_elements).find_by!(url: params[:page_id])
+    @element_grid = initialize_grid(page.page_elements)
+  end
+
+  def new
+    page = Page.includes(:page_elements).find_by!(url: params[:page_id])
+    @page_element = page.page_elements.build
+  end
+
+  def edit
+    page = Page.includes(:page_elements).find_by!(url: params[:page_id])
+    @page_element = page.page_elements.find(params[:id])
+  end
```

Comment:
Hmm.
**1**
In the `new`-action the `@page_element` is set as a new `PageElement` with `page_id` of the current page.

In the `edit`-action the `@page_element` is set to given ID if it has the right `page_id`.
I cannot see them being the same.

**2**

If `@page_element = page.page_elements.build` would trash page_elements?